fuzz/fuzz_targets/roundtrip.rs

```diff
@@ -24,12 +24,73 @@ fuzz_target!(|data: &[u8]| {
         Ok(bytes) => bytes,
         Err(_) => return,
     };
+
+    // Only roundtrip valid modules for now since invalid modules can often have
+    // bizarre structures which aren't intended to print correctly or roundtrip
+    // well.
     if wasmparser::validate(&wasm).is_err() {
         return;
     }
-    let string = match wasmprinter::print_bytes(&wasm) {
+
+    // And finally validate that the name section, if present, is valid. This
+    // can be invalid if names in the name section are too long (e.g. exceeding
+    // the maximum length of a string). The printing process will skip invalid
+    // name sections, so if it's invalid then our roundtrip'd bytes will
+    // trivially not match, but not in an interesting way.
+    if name_section(&wasm).is_err() {
+        return;
+    }
+    let string2 = match wasmprinter::print_bytes(&wasm) {
         Ok(s) => s,
         Err(_) => return,
     };
-    assert_eq!(wasm, wat::parse_str(&string).unwrap());
+
+    let wasm2 = wat::parse_str(&string2).unwrap();
+    if wasm == wasm2 {
+        return;
+    }
+
+    std::fs::write("wasm1.wasm", &wasm).unwrap();
+    std::fs::write("wasm1.wat", &string).unwrap();
+    std::fs::write("wasm2.wasm", &wasm2).unwrap();
+    std::fs::write("wasm2.wat", &string2).unwrap();
+    panic!("wasm bytes differ on roundtrip");
 });
+
+fn name_section(wasm: &[u8]) -> wasmparser::Result<()> {
```

Comment:
I would expect, based on the name, that this return the contents of the parsed name section. Suggest naming this `validate_name_section` or something like that.